enhancements/configurable-dev-env.md

```diff
@@ -0,0 +1,409 @@
+- **IPv6**: `/64` subnet
+
+All interfaces connected to the same switch receive IPs from the same subnet. This supports scenarios like the `leafkind-switch` which connects multiple nodes (leafkind, pe-kind-control-plane, pe-kind-worker).
+
+### Special Allocations
+
+Additional IP ranges may be reserved for:
+
+- **VTEP addresses**: Loopback IPs for VXLAN tunnel endpoints
+- **Router IDs**: BGP router identifiers (typically IPv4 addresses)
+- **Management**: Out-of-band management interfaces (if needed)
+
+## Automatic Property Assignment
+
+The system automatically assigns variable properties that would otherwise require manual specification for each node:
+
+### VTEP IPs (for edge leaves)
+- Allocated from a dedicated loopback range (e.g., `10.0.1.0/24`)
+- One unique IP per edge leaf node
+- Used as the source IP for VXLAN tunnels
+
+### MAC Addresses
+- Generated deterministically or randomly for VXLAN interfaces
+- Ensures uniqueness across the topology
+- Format: Locally administered MAC addresses (e.g., `02:xx:xx:xx:xx:xx`)
+
+### BGP Router IDs
+- Typically derived from VTEP IPs or allocated from a separate range
+- One per BGP-enabled router
+- Must be unique across all BGP speakers
+
+## Configuration Introspection API
+
+Since most configuration is now automatically generated, a query interface is essential for both operational use and testing.
+
+### CLI Interface
+
+A command-line tool provides query capabilities for common operations:
+
+```bash
+# Get VTEP IP for a specific leaf
+$ clab-config query --node leafA --field vtep-ip
+10.0.1.1
+
+# Get interface IP for a node towards another node
+$ clab-config query --node leafA --peer spine --ip-family ipv4
+192.168.0.1
+
+# Reverse lookup: Find which interface/link an IP belongs to
+$ clab-config query --ip 192.168.0.1
+Node: leafA
+Interface: eth1
+Peer: spine
+Link: leafA:eth1 <-> spine:eth1
+
+# Get all VRF information for a node
+$ clab-config query --node leafA --vrfs
+VRF: red
+  VNI: 100
+  Interfaces: [ethred]
+  VTEP: 10.0.1.1
+VRF: blue
+  VNI: 101
+  Interfaces: [ethblue]
+  VTEP: 10.0.1.1
+
+```
+
+### Go API
+
+For programmatic access (especially in e2e tests):
+
+```go
+// Load topology configuration
+topo, err := config.Load("topology-config.yaml")
+if err != nil {
+    log.Fatal(err)
+}
+
+// Query node properties
+vtepIP := topo.GetNodeVTEP("leafA")
+linkIP := topo.GetLinkIP("leafA", "spine", config.IPv4)
+
+// Get all nodes matching a pattern
+leaves := topo.GetNodesByPattern("leaf.*")
+
+// Reverse lookup
+link, iface := topo.FindIPOwner("192.168.0.1")
+```
+
+### Use Cases
+
+This introspection capability enables:
+
+1. **E2E tests**: Dynamically discover topology parameters without hardcoding
+2. **Debugging**: Quickly identify which node/interface owns an IP address
+3. **Documentation**: Generate topology diagrams and documentation automatically
+4. **Validation**: Verify configuration correctness before deployment
```

Comment:
is this a dry-run mode ? OR, are we actually loading the configs into an frr router (which version ? it would need to be configurable), get the running-conf from it, then match for correctness ?